logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserModel.kt

```diff
@@ -42,6 +42,7 @@ sealed class User {
     abstract val previewPicture: UserAssetId?
     abstract val completePicture: UserAssetId?
     abstract val availabilityStatus: UserAvailabilityStatus
+    abstract val supportedProtocols: Set<SupportedProtocol>?
```

Comment:
I wonder how to separate this once it comes from the database.

I assume it should come from the database at some point.

How to differentiate `null` from an empty set, or... if there should be any distinction. Can't we assume that an empty set is the same as "unknown" and make this not-optional ?

Reply:
Is this field intended to some sort of backwards compatibility ? (ie: proteus) 
I'm asking if in this case then makes sense to use Proteus as default value 🤔

Reply:
Yes, we can probably make it non-optional. Was initially thinking I wanted to keep it nullable to know if we had fetched it all or not. But `supportedProtocols` can always change so we anyway need to refresh regularly so it doesn't matter.

Reply:
> Is this field intended to some sort of backwards compatibility ? (ie: proteus)
> I'm asking if in this case then makes sense to use Proteus as default value 🤔

Going forward we will in a world were some users will support proteus and some MLS and some will support both. This field is used to announce to other users which protocols you support with your clients so that we can pick the right protocol.

We are defaulting proteus if a user has never set which protocols he/her support but we can't always default proteus before we have even attempted to fetch the supported protocols (the user might not support proteus).